AzureMaps/AzureMaps.php

```diff
@@ -0,0 +1,312 @@
+     * @return Collection
+     *
+     * @throws \Geocoder\Exception\Exception
+     */
+    public function geocodeQuery(GeocodeQuery $query): Collection
+    {
+        $url = $this->buildGeocodeUrl(self::GEOCODE_ENDPOINT_SSL, $query->getText());
+
+        $content = $this->getUrlContents($url);
+
+        $response = $this->validateResponse($content, $url);
+        $addresses = $this->formatGeocodeResponse($response);
+
+        return new AddressCollection($addresses);
+    }
+
+    /**
+     * @param ReverseQuery $query
+     *
+     * @return Collection
+     *
+     * @throws \Geocoder\Exception\Exception
+     */
+    public function reverseQuery(ReverseQuery $query): Collection
+    {
+        $url = $this->buildReverseGeocodeUrl(
+            self::REVERSE_ENDPOINT_URL,
+            $query->getCoordinates()->getLatitude(),
+            $query->getCoordinates()->getLongitude()
+        );
+
+        $content = $this->getUrlContents($url);
+
+        $response = $this->validateResponse($content, $url);
+        $addresses = $this->formatReverseGeocodeResponse($response);
+
+        return new AddressCollection($addresses);
+    }
+
+    /**
+     * Returns the provider's name.
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        return 'azure_maps';
+    }
+
+    /**
+     * Returns an array of non null geocode /reverse-geocode options
+     * @param array $options
+     */
+    private function setOptions(array $options)
+    {
+        $options = array_merge($this->options, $options);
+
+        $this->options = array_filter($options, function ($option) {
+            return !is_null($option);
+        });
+    }
+
+    /**
+     * Returns an array of keys to replace
+     *
+     * @return array
+     */
+    public function getOptions(): array
+    {
+        return $this->options;
+    }
+
+    /**
+     * @param string $url
+     * @param string $query
+     * @return string
+     */
+    private function buildGeocodeUrl(string $url, string $query): string
+    {
+        $format = $this->getFormat();
+        $options = http_build_query($this->getOptions());
+        $query = urlencode($query);
+        $subscriptionKey= $this->getSubscriptionKet();
+
+        return "{$url}/$format?api-version=1.0&subscription-key={$subscriptionKey}&{$options}&query={$query}";
+    }
+
+    /**
+     * @param string $url
+     * @param string $latitude
+     * @param string $longitude
+     * @return string
+     */
+    private function buildReverseGeocodeUrl(string $url, string $latitude, string $longitude): string
+    {
+        $format = $this->getFormat();
+        $options = http_build_query($this->getOptions());
+        $subscriptionKey= $this->getSubscriptionKet();
+
+        return "{$url}/$format?api-version=1.0&subscription-key={$subscriptionKey}&{$options}&query={$latitude},{$longitude}";
+    }
+
+    /**
+     * @return string
+     */
+    private function getSubscriptionKet(): string
```

Comment:
typo in the name
```suggestion
    private function getSubscriptionKey(): string
```

Reply:
What is the use of this method?

Reply:
$subscriptionKey is a property seperated from the $options property, therefor it has a getter. Being called on AzureMaps/AzureMaps.php#170

Reply:
We don't need a setter/getter for an internal property. Just set it to a `protected` property :)